src/libServer/GetworkServer.cpp

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (C) 2019 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <chrono>
+
+#include <jsonrpccpp/server/connectors/httpserver.h>
+#include "depends/libethash/include/ethash/ethash.hpp"
+
+#include "GetworkServer.h"
+#include "common/Constants.h"
+#include "libPOW/pow.h"
+
+using namespace std;
+using namespace jsonrpc;
+
+//////////////////////////////////////////////////
+// Helper functions
+//////////////////////////////////////////////////
+
+static ethash_mining_result_t FAIL_RESULT = {"", "", 0, false};
+
+bool HexStringToUint64(const std::string& s, uint64_t* res) {
```

Comment:
HexStringToUint64 and NormalizeHexString can add to DataConversion class. And add a few unit test in Test_DataConversion.cpp

Reply:
HexStringToUint64 can implemented as a more general template function HexStringToIngeter? If it is too hard, then just leave it.